apps/jobs/views.py

```diff
@@ -624,6 +624,149 @@ def leaderboard(request, challenge_phase_split_id):
     return paginator.get_paginated_response(response_data)
 
 
+@api_view(["GET"])
+@throttle_classes([AnonRateThrottle])
+def complete_leaderboard(request, challenge_phase_split_id):
+    """
+    Returns complete leaderboard for a corresponding Challenge Phase Split
+
+    - Arguments:
+        ``challenge_phase_split_id``: Primary key for the challenge phase split for which leaderboard is to be fetched
+
+    - Returns:
+        All Leaderboard entry objects in a list
+    """
+    # check if the challenge exists or not
+    try:
+        challenge_phase_split = ChallengePhaseSplit.objects.get(
+            pk=challenge_phase_split_id
+        )
+    except ChallengePhaseSplit.DoesNotExist:
+        response_data = {"error": "Challenge Phase Split does not exist"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Please replace the above try/except block with 
```
    challenge_phase_split = get_challenge_phase_split_model(challenge_phase_split_pk)
```
and also add 
```
get_challenge_phase_split_model = get_model_object(ChallengePhaseSplit)
```
in jobs/utils.py at line 21 just above the `get_submission_model = get_model_object(Submission)`